test/e2e/common/downward_api.go

```diff
@@ -220,3 +220,214 @@ func testDownwardAPI(f *framework.Framework, podName string, env []v1.EnvVar, ex
 func testDownwardAPIUsingPod(f *framework.Framework, pod *v1.Pod, env []v1.EnvVar, expectations []string) {
 	f.TestContainerOutputRegexp("downward api env vars", pod, 0, expectations)
 }
+
+func downwardAPIVolumePodForModeTest(name, filePath string, itemMode, defaultMode *int32) *v1.Pod {
```

Comment:
This method should be in the `*_volume.go` file right?

Reply:
should I move all the helper functions into the respective file that uses it? I wasn't sure whether I should do that or just keep them all in the "base" (non-volume) file.

Reply:
Ok so I've moved all helper functions based on following criterea:

If used in [testname]_volume.go exclusively, file is moved to [testname]_volume.go
Else moved to [testname].go

I won't update the PR until I get a confirmation that this is a desired change.